packages/protocol/contracts/authorizers/RankedAdminnable.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./interfaces/IRankedAdminnable.sol";
+
+/// @title Contract that implements multiple ranks of admins
+contract RankedAdminnable is IRankedAdminnable {
+  /// @notice Keeps the ranks of admins for each individual adminned entity
+  /// @dev This contract implements adminship for a mapping of entities, rather
+  /// than a single entity. However, the logic at the inheriting contract can
+  /// easily be adapted to use this contract to admin a single entity (see
+  /// Api3CwRrpAuthorizer.sol)
+  mapping(bytes32 => mapping(address => uint256)) public adminnedIdToAdminToRank;
+
+  /// @dev Reverts if the caller's rank is not greater than or equal to `rank`
+  /// @param adminnedId ID of the entity being adminned
+  /// @param rank Rank caller's rank will be compared to
+  modifier onlyWithRank(bytes32 adminnedId, uint256 rank) {
+    require(getRank(adminnedId, msg.sender) >= rank, "Caller is ranked low");
```

Comment:
having thought about it, due to the way numbers are initialized  from 0 its probably harder to do lower rank number - higher precedence because everything will have been initialized with highest rank which will be a pain to workaround.